service/header/interface.go

```diff
@@ -0,0 +1,39 @@
+package header
+
+import (
+	"context"
+
+	tmbytes "github.com/celestiaorg/celestia-core/libs/bytes"
+)
+
+// Subscriber encompasses the behavior necessary to
+// subscribe/unsubscribe from new ExtendedHeader events from the
+// network.
+type Subscriber interface {
+	Subscribe() (Subscription, error)
+}
+
+// Subscription can retrieve the next ExtendedHeader from the
+// network.
+type Subscription interface {
+	NextHeader(ctx context.Context) (*ExtendedHeader, error)
```

Comment:
It's not entirely clear to me how the actual implementation will look like. 
I saw this: [In headerSub, we will have a subscription that allows you to do `NextHeader` (that will return the next header from the gossipsub network).](https://github.com/celestiaorg/celestia-node/pull/148#discussion_r740197101). So I can somewhat guess (gossipsub will fill some kind of queue in the background which can be read from via this subscribe?)

Reply:
Yes

Reply:
Yes, so `Subscription` will abstract the behaviour for reading the next header from the gossipsub network. Many subscriptions can exist. The nice part about this is several parts of the codebase may want to listen for new headers, so each can use its own subscription. This doesn't add any overhead in terms of p2p (as @Wondertan explained), since creating a new Subscription doesn't register as a new peer joining the topic. It's all on the same host.

Reply:
@renaynay, pls add the comment that it returns only verified and valid headers. Currently, we don't have validation but we will add with #157. It is important to note that.